gateway/src/apicast/configuration_loader/remote_v2.lua

```diff
@@ -107,44 +176,39 @@ function _M:index(host)
   ngx.log(ngx.DEBUG, 'index get status: ', res.status, ' url: ', url)
 
   if res.status == 200 then
-    local json = cjson.decode(res.body)
-
-    local config = { services = array(), oidc = array() }
-
-    local proxy_configs = json.proxy_configs or {}
-
-    for i=1, #proxy_configs do
-      local proxy_config = proxy_configs[i].proxy_config
-
-      -- Copy the config because parse_service have side-effects. It adds
-      -- liquid templates in some policies and those cannot be encoded into a
-      -- JSON. We should get rid of these side effects.
-      local original_proxy_config = deepcopy(proxy_config)
-
-      local service = configuration.parse_service(proxy_config.content)
-      local oidc = self:oidc_issuer_configuration(service)
+    return parse_resp_body(self, res.body)
+  else
+    return nil, 'invalid status'
+  end
+end
 
-      -- Assign false instead of nil to avoid sparse arrays. cjson raises an
-      -- error by default when converting sparse arrays.
-      config.oidc[i] = oidc or false
+function _M:load_configs_for_env_and_host(env, host)
+  local url = endpoint_for_services_with_host(self.endpoint, env, host)
 
-      config.services[i] = original_proxy_config.content
-    end
+  local response = self.http_client.get(url)
 
-    return cjson.encode(config)
+  if response.status == 200 then
+    return parse_resp_body(self, response.body)
   else
-    return nil, 'invalid status'
+    ngx.log(ngx.ERR, 'failed to load proxy configs')
+    return false
   end
 end
 
 function _M:call(environment)
+  local load_just_for_host = load_just_the_services_needed()
+
   if self == _M  or not self then
     local host = environment
     local m = _M.new()
     local ret, err = m:index(host)
 
     if not ret then
-      return m:call()
+      if load_just_for_host then
```

Comment:
Does not make sense to me to keep the negative condition here, I think that can be something like this: 

```
if ret then
return ret, err
end

if load_just_for_host then
...
```

I think that makes things more clear.

Reply:
You're absolutely right. I've simplified this :+1: